src/AutoRest.CSharp/Output/Models/Types/OutputLibrary.cs

```diff
@@ -305,5 +305,102 @@ private static string ConstructResourceType(string httpRequestUri)
             }
             return null;
         }
+
+        public bool IsTenantOnly(List<List<ProviderToken>> tokens, string providerName)
+        {
+            bool foundTenant = false;
+            bool foundNonTenant = false;
+            for (int j = 0; j < tokens.Count && (!foundTenant || !foundNonTenant); j++)
+            {
+                var tokenList = tokens[j];
+                for (int i = 0; i < tokenList.Count && (!foundTenant || !foundNonTenant); i++)
+                {
+                    var token = tokenList[i];
+                    var operationMatch = VerifyOperation(token.tokenValue, providerName);
+                    foundNonTenant = !foundNonTenant ? token.isFullProvider && !token.noPredecessor && operationMatch : true;
+                    foundTenant = !foundTenant ? token.isFullProvider && token.noPredecessor && operationMatch : true;
+                }
+            }
+            return foundTenant && !foundNonTenant;
+        }
+
+        //Extensions algo will use same tokens ADO #5523
+        public static List<ProviderToken> Tokenize(string path)
```

Comment:
I feel like this method and MakeTokens belong to a different class that OutputLibrary can use